lima-network.awk

```diff
@@ -0,0 +1,99 @@
+# This script collects information from `ip -0 link` about current interface
+# names with data from "cidata/network-config" about the desired state.
+#
+# Arrays:
+# - curname[addr] => current name for addr
+# - use[itf]      => current address for interface (name is "in use" if recorded here)
+# - newname[addr] => desired name for addr
+#
+# The script tries to minimize the number of renames. It processes as many as possible,
+# and if newnames is not yet empty, it renames one of the addresses to a temporary
+# name to break cyclic dependencies.
+#
+# The generated script is being running by lima-init when all links are still down, so
+# no commands are produced to change the interface status (names can only be changed
+# in DOWN status).
+
+### Collect info from `ip -o link`.
+# 3: eth1: <BROADCAST,MULTICAST> [...] state DOWN qlen 1000\    link/ether 22:aa:56:ca:ea:6e [...]
+/link\/ether/ {
```

Comment:
If you are using ip command from iproute2, you can use ip -j to get JSON output

Reply:
busybox `ip` does not support it. But this will become moot once the logic is re-implemented in Go.